services/nexus/nexus.service.js

```diff
@@ -27,20 +27,30 @@ const searchApiSchema = Joi.object({
     .required(),
 }).required()
 
-const resolveApiSchema = Joi.object({
+const nexus3SearchApiSchema = Joi.object({
+  items: Joi.array()
+    .items(
+      Joi.object({
+        // This schema is relaxed similarly to nexux2SearchApiSchema
+        version: Joi.string().required(),
+      })
+    )
+    .required(),
+}).required()
+
+const nexus2ResolveApiSchema = Joi.object({
   data: Joi.object({
     baseVersion: optionalDottedVersionNClausesWithOptionalSuffix,
     version: optionalDottedVersionNClausesWithOptionalSuffix,
   }).required(),
 }).required()
 
-// https://repository.sonatype.org/nexus-restlet1x-plugin/default/docs/path__artifact_maven_resolve.html
-// https://repository.sonatype.org/nexus-indexer-lucene-plugin/default/docs/path__lucene_search.html
 const queryParamSchema = Joi.object({
   server: optionalUrl.required(),
   queryOpt: Joi.string()
-    .regex(/(:(?:q|g|a|v|p|c|cn|sha1|from|count|repositoryId|e|r)=[\w-. ]+)+/i)
+    .regex(/(:[\w.]+=[\w-. ]+)+/i)
```

Comment:
The Nexus 3 search endpoint already supports 35 different parameters. I decided to loosen the validation rather than coming up with a ridiculously long regex!